sklego/preprocessing.py

```diff
@@ -99,19 +98,20 @@ def transform(self, X):
         Transforms pandas dataframe by (de)selecting columns based on their dtype
         :param X: pandas dataframe to select dtypes for
         """
-        check_is_fitted(self, ['type_columns_', 'X_dtypes_', 'feature_names_'])
-        if (self.X_dtypes_ != X.dtypes).any():
-            raise ValueError(f'Column dtypes were not equal during fit and transform. Fit types: \n'
-                             f'{self.X_dtypes_}\n'
-                             f'transform: \n'
-                             f'{X.dtypes}')
+        check_is_fitted(self, ['X_dtypes_', 'feature_names_'])
+
+        try:
+            if (self.X_dtypes_ != X.dtypes).any():
+                raise ValueError(f'Column dtypes were not equal during fit and transform. Fit types: \n'
+                                 f'{self.X_dtypes_}\n'
+                                 f'transform: \n'
+                                 f'{X.dtypes}')
+        except ValueError:
+            raise ValueError(f'Columns were not equal during fit and transform')
```

Comment:
We probably want to do this:

```
except ValueError as e:
    raise ValueError(f'Columns were not equal during fit and transform') from e
```

So that the stacktrace will have additional info

Reply:
@MBrouns I handled it now.